paho/client.go

```diff
@@ -333,6 +342,24 @@ func (c *Client) Connect(ctx context.Context, cp *Connect) (*Connack, error) {
 	return ca, nil
 }
 
+func (c *Client) waitWrite() {
+	for {
+		wwt, ok := <-c.outPackets
+		if !ok {
+			return
+		}
+		_, wwt.e = wwt.w.WriteTo(c.Conn)
+	}
+}
+
+func (c *Client) doWrite(sp io.WriterTo) error {
+	wt := &whatWriteTo{
+		w: sp,
+	}
+	c.outPackets <- wt
+	return wt.e
```

Comment:
with uncached chan, errors are not lost. 
```
outPackets:   make(chan *whatWriteTo)
```

```
func (c *Client) waitWrite() {
	for {
		wwt, ok := <-c.outPackets
		if !ok {
			return
		}
		_, wwt.e = wwt.w.WriteTo(c.Conn)
	}
}
```

Reply:
They can be, because nothing waits for `wwt.w.WriteTo(c.Conn)` to finish. 

Playground example, notice `out` will never be printed out: https://play.golang.com/p/uTr1ShpRmoD

Reply:
but, waitWrite() no time.Sleep(1).

Reply:
fix:  https://go.dev/play/p/uzBqjFo29jR